resources/server_install.rb

```diff
@@ -49,21 +49,30 @@
 
   log 'Enable and start MariaDB service' do
     notifies :enable, "service[#{platform_service_name}]", :immediately
+    notifies :stop, "service[#{platform_service_name}]", :immediately
+    notifies :create, 'file[generate-mariadb-root-password]', :immediately
+    notifies :run, 'execute[apply-mariadb-root-password]', :immediately
     notifies :start, "service[#{platform_service_name}]", :immediately
   end
 
   mariadb_root_password = new_resource.password == 'generate' || new_resource.password.nil? ? secure_random : new_resource.password
 
-  # Generate a ramdom password or set the a password defined with node['mariadb']['password']['root'].
+  # Generate a random password or set the a password defined with node['mariadb']['server_root_password'].
   # The password is set or change at each run. It is good for security if you choose to set a random password and
   # allow you to change the root password if needed.
-  bash 'generate-mariadb-root-password' do
-    user 'root'
-    code <<-EOH
-    echo "ALTER USER 'root'@'localhost' IDENTIFIED BY \'#{mariadb_root_password}\';" | /usr/bin/mysql
-    EOH
+  file 'generate-mariadb-root-password' do
+    owner 'root'
+    group 'root'
+    mode '600'
+    sensitive true
+    content "ALTER USER 'root'@'localhost' IDENTIFIED BY '#{mariadb_root_password}';"
```

Comment:
Hey, maybe I am missing something but I don't see where we run this code, where as before we were executing it in bash?

Reply:
My bad, I'll resubmit the PR, has I probably deleted the path line, shameful =/

Reply:
its fine, just push another commit, these things happen :) 
I think what you want is just to run a bash with `cat #{file} | mysqld` ?